panel/io/server.py

```diff
@@ -76,9 +76,12 @@ def unlocked():
         events = []
         for conn in connections:
             socket = conn._socket
+            if hasattr(socket, 'write_lock') and socket.write_lock._block._value == 0:
+                state._locks.append(id(socket))
+            locked = id(socket) in state._locks
```

Comment:
btw. This seems to me like a perfect case for using `WeakSet`.

Reply:
👍